feature_engine/utils/calculate_conversation_level_features.py

```diff
@@ -11,281 +11,136 @@
-		)
-
-		self.conv_data = pd.merge(
-			left=self.conv_data,
-			right=get_min(self.chat_data, 'info_exchange_zscore_conversation', 'min_info_exchange_zscore_conversation'),
-			on=["conversation_num"],
-			how="inner"
-		)
+        """
+        # Get gini based features
+        self.get_gini_features()
+        # Get summary statistics by aggregating chat level features
+        self.get_conversation_level_summary_statistics_features()
+        self.get_talkative_member_features()
+
+        # TODO - currently, lexical features are not being summarized!
+
+        return self.conv_data
+
+    def get_gini_features(self) -> None:
+        """
+            This function is used to calculate the gini index for each conversation 
+            based on the word level and character level information.
+        """
+        # Gini for #Words
+        self.conv_data = pd.merge(
+            left=self.conv_data,
+            right=get_gini(self.chat_data, "num_words"),
+            on=['conversation_num'],
+            how="inner"
+        )
+        # Gini for #Characters
+        self.conv_data = pd.merge(
+            left=self.conv_data,
+            right=get_gini(self.chat_data, "num_chars"),
+            on=['conversation_num'],
+            how="inner"
+        )
+
+    def get_conversation_level_summary_statistics_features(self) -> None:
+        """
+            This function is used to aggregate the summary statistics from 
+            chat level features to conversation level features.
+            Specifically, it looks at the mean and standard deviations at message and word level.
+        """
+        # For each summarizable feature
+        for column in self.columns_to_summarize:
+            # Average/Mean of feature across the Conversation
+            self.conv_data = pd.merge(
+                left=self.conv_data,
+                right=get_average(self.chat_data, column, 'average_'+column),
+                on=['conversation_num'],
+                how="inner"
+            )
+
+            # Standard Deviation of feature across the Conversation
+            self.conv_data = pd.merge(
+                left=self.conv_data,
+                right=get_stdev(self.chat_data, column, 'stdev_'+column),
+                on=['conversation_num'],
+                how="inner"
+            )
+
+            # Minima for the feature across the Conversation
+            self.conv_data = pd.merge(
+                left=self.conv_data,
+                right=get_min(self.chat_data, column, 'min_'+column),
+                on=['conversation_num'],
+                how="inner"
+            )
+
+            # Maxima for the feature across the Conversation
+            self.conv_data = pd.merge(
+                left=self.conv_data,
+                right=get_max(self.chat_data, column, 'max_'+column),
+                on=['conversation_num'],
+                how="inner"
+            )
+
+    def get_talkative_member_features(self) -> None:
```

Comment:
Note: the columns generated by this function are redundant, given that you are summarizing all numeric columns. I'll delete this function in a later commit.

Reply:
(Resolved!)

Reply:
Thanks again for the quick fix!